### PR TITLE
chore: update kotlin to 1.8.22; kotlinx.coroutines to 1.7.3

### PR DESCRIPTION
**Pull request by aajtodd**
## Issue \#

N/A

## Description of changes

* Update Kotlin to 1.8.22
* Update kotlinx.coroutines to 1.7.3

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by 0marperez — APPROVED** [review]

**Review by lauzadis — APPROVED** [review]
1.8.22 is not available internally (latest is 1.8.21). Will that affect internal usage of the SDK?

**Comment by aajtodd** [comment]
@lauzadis 

> 1.8.22 is not available internally (latest is 1.8.21). Will that affect internal usage of the SDK?

Looking at the [release notes](https://github.com/JetBrains/kotlin/releases/tag/v1.8.22) I don't think so.

File: aws-runtime/build.gradle.kts [hunk 1]
```diff
@@ -86,28 +86,6 @@ subprojects {
     }
 }
 
-task<org.jetbrains.kotlin.gradle.testing.internal.KotlinTestReport>("rootAllTest") {
```
Comment (antohaby) [inline aws-runtime/build.gradle.kts:89]: Hey! Kotlin Gradle Plugin developer here. 
Can you guys confirm that you don't need this code anymore? 
We have this bug that we want to close as "Won't Fix". But instead we want to improve UX here. 
Would you be so kind sharing your Use Case? 

Thanks

Comment (lauzadis) [reply aws-runtime/build.gradle.kts:89]: Hi, the code you are commenting on was deleted in this PR almost 1 year ago, and we never added it back, so it's safe to say we don't need it anymore.

**Pull request merged**